File: ukidelly/PGS/240608/홀수 vs 짝수.java
class Solution {
  public int solution(int[] numList) {
    int evenSum = 0;
    int oddSum = 0;

    for(int idx=0; idx<numList.length; idx++) {
      if(idx%2==0) {
        oddSum += numList[idx];
      }
      else {
        evenSum += numList[idx];
      }
    }

    return Math.max(evenSum,oddSum);
  }
}

File: ukidelly/PGS/240608/홀수 vs 짝수.kt
import kotlin.math.max

class Solution {
  fun solution(numList: IntArray): Int {

    val evenSum = numList.filterIndexed { index, _ -> index % 2 == 1 }.sum()
    val oddSum = numList.filterIndexed { index, _ -> index % 2 == 0 }.sum()

    return max(evenSum, oddSum)
  }
}